### PR TITLE
security audits 2

### PR DESCRIPTION
**Pull request by chaosma**
* remove unnecessary condition check in incrementalQuinnTree.circom
* added SafeLessThan, SafeGreaterThan circuit for safety use comparator circuits
* replaced LessThan/GreaterThan usages by Safe ones. This includes security audits items as well as some other places. Only except is float.circom it involves too much computation for subsidy.

### CODE REVIEW

File: circuits/circom/trees/incrementalQuinTree.circom [hunk 1]
```diff
@@ -280,10 +284,6 @@ template QuinGeneratePathIndices(levels) {
     n[levels] <-- m;
 
     // Do a range check on each out[i]
```
Comment (gurrpi) [inline circuits/circom/trees/incrementalQuinTree.circom:286]: Don't forget to remove comments!

Comment (chaosma) [reply circuits/circom/trees/incrementalQuinTree.circom:286]: thanks for reminding!

**Comment by gurrpi** [comment]
Blocked by https://github.com/privacy-scaling-explorations/maci/pull/659.
Also there is an issue on CI side when running test on `maci-circuits`.

**Comment by chaosma** [comment]
> Blocked by #659. Also there is an issue on CI side when running test on `maci-circuits`.

I merged #659. Will wait for the CI issue to solve.

**Pull request merged**